### DISCUSSION TITLE
woocommerce_sale_flash filter not respected

### DISCUSSION THREAD

**Discussion by aheckler** [woocommerce/woocommerce#45041] [WooCommerce, Blocks, and Site Editing]
## Describe the bug

Any code that filters on `woocommerce_sale_flash` expecting to change the sale badge doesn't work with the blocks in this plugin.

## To reproduce

1. Add this custom code to your site to make the sale badge show `-30%` or whatever the discount is:
```
function my_custom_sale_flash($text) {
    global $product;
    $percentage = round( ( ( $product->regular_price - $product->sale_price ) / $product->regular_price ) * 100 );
    return '<span class="onsale">-'.$percentage.'%</span>';  
}
add_filter('woocommerce_sale_flash', 'my_custom_sale_flash');
```
2. Create an on-sale product.
3. View the product on the regular shop page. You'll see the sale badge show `-30%` or whatever.
4. Create a new page and use the All Products block, publish it.
5. View that page and find the test product. The badge still says just `Sale`.

## Expected behavior

The badge should respect any filtering on `woocommerce_sale_flash` I would think.

**Comment by nerrad** [comment]
Hi @aheckler, thanks for leaving your feedback! Currently the All Products block uses a different method of rendering content than the other pages in Woo. The content is generated client side with React components. Thus extending the way different aspects of the content displays cannot be modified via existing filters and actions like you are accustomed here.

We are currently collecting posts/requests like what you shared here to help us guide how we expose this extensibility/customization for future iterations of the blocks. As an example we could enhance the sale badge block so that there are more customization options for what is available for display in the badge (and potential data sources).

Since this is not currently actionable - I'm closing this issue but rest assured what you've taken the time to write here is something noted.
